core/include/atomic.h

```diff
@@ -132,6 +132,44 @@ static inline int atomic_set_to_zero(atomic_int_t *var)
 }
 
 /**
+ * @brief  Set var to (var & val) atomically and return the old value.
+ *
+ * @param[inout]  var   Pointer to atomic variable
+ * @param[in]     val   Value to AND
+ *
+ * @return The value of *var* before the operation.
+ */
+static inline int atomic_and(atomic_int_t *var, int val)
+{
+    int old;
+
+    do {
+        old = var->value;
+    } while (!atomic_cas(var, old, old & val));
+
+    return old;
+}
+
+/**
+ * @brief  Set var to (var | val) atomically and return the old value.
+ *
+ * @param[inout]  var   Pointer to atomic variable
+ * @param[in]     val   Value to OR
+ *
+ * @return The value of *var* before the operation.
+ */
+static inline int atomic_or(atomic_int_t *var, int val)
```

Comment:
Maybe make this file a PR on its own, and also add `xor` and `not` to complete the family?